File: app/src/main/java/com/fastaccess/provider/timeline/handler/drawable/UrlDrawable.java
package com.fastaccess.provider.timeline.handler.drawable;

import android.graphics.Canvas;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;

class UrlDrawable extends BitmapDrawable {
    private Drawable drawable;

    @SuppressWarnings("deprecation") UrlDrawable() {}

    @Override public void draw(Canvas canvas) {
        if (drawable != null) {
            drawable.draw(canvas);
        }
    }

    public Drawable getDrawable() {
        return drawable;
    }

    public void setDrawable(Drawable drawable) {
        this.drawable = drawable;
    }
}

File: markdown/src/main/java/com/fastaccess/markdown/spans/drawable/UrlDrawable.kt
package com.fastaccess.markdown.spans.drawable

import android.graphics.Canvas
import android.graphics.drawable.BitmapDrawable
import android.graphics.drawable.Drawable

import com.bumptech.glide.load.resource.gif.GifDrawable

class UrlDrawable : BitmapDrawable(), Drawable.Callback {

    private var drawable: Drawable? = null

    override fun draw(canvas: Canvas) {
        drawable?.let {
            it.draw(canvas)
            if (it is GifDrawable) {
                if (!it.isRunning) it.start()
            }
        }
    }

    override fun invalidateDrawable(who: Drawable) {
        if (callback != null) {
            callback!!.invalidateDrawable(who)
        }
    }

    override fun scheduleDrawable(who: Drawable, what: Runnable, `when`: Long) {
        if (callback != null) {
            callback!!.scheduleDrawable(who, what, `when`)
        }
    }

    override fun unscheduleDrawable(who: Drawable, what: Runnable) {
        if (callback != null) {
            callback!!.unscheduleDrawable(who, what)
        }
    }

    fun setDrawable(drawable: Drawable) {
        this.drawable?.callback = this
        drawable.callback = this
        this.drawable = drawable
    }
}